core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLBatchTestBase.java

```diff
@@ -178,4 +190,52 @@ protected void cleanPartitions(TimePartitionedFileSet fileSet) throws IOExceptio
     }
     return records;
   }
+
+  protected ApplicationManager deployETL(ETLBatchConfig etlConfig, String appName) throws Exception {
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app(appName);
+    return deployApplication(appId, appRequest);
+  }
+
+  /**
+   * Run the SmartWorkflow in the given ETL application for once and wait for the workflow's COMPLETED status
+   * with 5 minutes timeout.
+   *
+   * @param appManager the ETL application to run
+   */
+  protected WorkflowManager runETLOnce(ApplicationManager appManager)
+    throws TimeoutException, InterruptedException, ExecutionException {
+    return runETLOnce(appManager, ImmutableMap.<String, String>of());
+  }
+
+  /**
+   * Run the SmartWorkflow in the given ETL application for once and wait for the workflow's COMPLETED status
+   * with 5 minutes timeout.
+   *
+   * @param appManager the ETL application to run
+   * @param arguments the arguments to be passed when running SmartWorkflow
+   */
+  protected WorkflowManager runETLOnce(ApplicationManager appManager, Map<String, String> arguments)
```

Comment:
this seems like a bad API - that if I want to pass in arguments, then I can't pass in a timeout.
And if I want to pass in a timeout, then I can't pass in arguments.

Reply:
I'm thinking maybe the API to set timeout is not necessary, since the default 5 min timeout should be sufficient.

Reply:
For unit tests, I agree.